src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceLifetime.cs

```diff
@@ -18,8 +18,10 @@ namespace Microsoft.Extensions.Hosting.WindowsServices
     public class WindowsServiceLifetime : ServiceBase, IHostLifetime
     {
         private readonly TaskCompletionSource<object?> _delayStart = new TaskCompletionSource<object?>(TaskCreationOptions.RunContinuationsAsynchronously);
+        private readonly TaskCompletionSource<object?> _serviceStopped = new TaskCompletionSource<object?>(TaskCreationOptions.RunContinuationsAsynchronously);
         private readonly ManualResetEventSlim _delayStop = new ManualResetEventSlim();
         private readonly HostOptions _hostOptions;
+        private bool _isStopped;
```

Comment:
The naming here between "isStopped" and "serviceStopped" is a little confusing.  What is the "is" referring to if not "service"?  I'm interpreting "isStopped" more as "stopRequested"?

Reply:
I was trying to represent what SCM thought about the service state.  I was running into a conflict in name between the task `serviceStopped` and this flag (I even inadvertently named them the same thing at first).  I'm open to other naming suggestions that would be clearer.

Maybe for the task use `serviceRunComplete` or `serviceDispatcherStopped` then for the flag use `serviceStopped`?

Reply:
I gave another try at the naming.  Will see if that's better.